Hcal/include/Hcal/HcalSimpleDigiAndRecProducer.h

```diff
@@ -0,0 +1,40 @@
+#ifndef HCALSIMPLEDIGIANDRECPRODUCER_H
+#define HCALSIMPLEDIGIANDRECPRODUCER_H
+#include "DetDescr/DetectorID.h"
+#include "DetDescr/HcalGeometry.h"
+#include "DetDescr/HcalID.h"
+#include "Framework/Configure/Parameters.h"
+#include "Framework/EventDef.h"
+#include "Framework/EventProcessor.h"
+#include "Framework/RandomNumberSeedService.h"
+#include "TRandom3.h"
+#include "Tools/NoiseGenerator.h"
+
+namespace hcal {
+class HcalSimpleDigiAndRecProducer : public framework::Producer {
+ public:
+  HcalSimpleDigiAndRecProducer(const std::string& name,
+                               framework::Process& process)
+      : framework::Producer{name, process} {}
+  ~HcalSimpleDigiAndRecProducer() override = default;
+  void configure(framework::config::Parameters& ps) override;
+  void SetupRandomNumberGeneration();
```

Comment:
shouldnt this be lower case, since it's a function, not a class?

Reply:
yep! and its content can be moved to `onNewRun` after a Framework patch awhile ago